org.eclipse.m2e.pde.connector/src/org/eclipse/m2e/pde/connector/TychoPackagingsConfigurator.java

```diff
@@ -0,0 +1,131 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Konrad Windszus
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *      Konrad Windszus
+ *******************************************************************************/
+package org.eclipse.m2e.pde.connector;
+
+import java.util.List;
+
+import org.apache.maven.plugin.MojoExecution;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.ProjectScope;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+import org.eclipse.m2e.core.project.configurator.AbstractProjectConfigurator;
+import org.eclipse.m2e.core.project.configurator.ProjectConfigurationRequest;
+import org.eclipse.pde.ds.internal.annotations.DSAnnotationVersion;
+import org.eclipse.pde.internal.core.natures.PDE;
+import org.eclipse.pde.internal.core.util.CoreUtility;
+import org.osgi.framework.Version;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@SuppressWarnings("restriction")
+public class TychoPackagingsConfigurator extends AbstractProjectConfigurator {
+
+    private static final String TYCHO_GROUP_ID = "org.eclipse.tycho";
+    private static final String TYCHO_DS_PLUGIN_ARTIFACT_ID = "tycho-ds-plugin";
+    private static final String GOAL_DECLARATIVE_SERVICES = "declarative-services";
+
+    private static final Logger log = LoggerFactory.getLogger(TychoPackagingsConfigurator.class);
+   
+    @Override
+    public void configure(ProjectConfigurationRequest request, IProgressMonitor monitor) throws CoreException {
+        MavenProject mavenProject = request.mavenProject();
+        IProject project = request.mavenProjectFacade().getProject();
+
+        String packaging = mavenProject.getPackaging();
+        if ("eclipse-plugin".equals(packaging) || "eclipse-test-plugin".equals(packaging)) {
+            PDEProjectHelper.configurePDEBundleProject(project, mavenProject, monitor);
+            applyDsConfiguration(request, monitor);
+        } else if ("eclipse-feature".equals(packaging)) {
+            // see org.eclipse.pde.internal.ui.wizards.feature.AbstractCreateFeatureOperation
+            if (!project.hasNature(PDE.FEATURE_NATURE)) {
+                CoreUtility.addNatureToProject(project, PDE.FEATURE_NATURE, monitor);
```

Comment:
```suggestion
                addNature(project, PDE.FEATURE_NATURE, monitor);
```
You could use the `addNature()` method of the super-class here. Looks like this does the same and is an API method in contrast to CoreUtility.

Reply:
Done in https://github.com/eclipse-m2e/m2e-core/pull/967/commits/b30f5013acbf8e7e38c8e6166de8e9ac876f278a.